pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -42,19 +42,22 @@ import (
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/dockerregistry/server"
 	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
+	"github.com/openshift/origin/pkg/dockerregistry/server/metrics"
 )
 
 // Execute runs the Docker registry.
 func Execute(configFile io.Reader) {
-	config, err := configuration.Parse(configFile)
+	dockerConfig, extraConfig, err := registryconfig.Parse(configFile)
 	if err != nil {
 		log.Fatalf("error parsing configuration file: %s", err)
 	}
-	setDefaultMiddleware(config)
-	setDefaultLogParameters(config)
+	setDefaultMiddleware(dockerConfig)
+	setDefaultLogParameters(dockerConfig)
 
 	ctx := context.Background()
-	ctx, err = configureLogging(ctx, config)
+	ctx = server.WithConfiguration(ctx, extraConfig)
```

Comment:
@miminar Please no. Otherwise we have to add that to the repository middleware options as well. It already contains many parameters. I don't want to adв internal options to repository middleware.
Please take a look at my new changes.